testing/MLDB-180-basic-join.js

```diff
@@ -251,4 +251,105 @@ expected = [
 testQuery('SELECT test1.* FROM test1 JOIN test2 ON test1.x = test2.x',
           expected);
 
+// MLDB-1189 OUTER JOINS
+
+expected = [
+   [ "_rowName", "test1.x", "test1.y", "test2.x", "test2.z" ],
+   [ "ex1-ex4", 1, 2, 1, 2 ],
+   [ "ex1-ex6", 1, 2, null, 3 ],
+   [ "ex1-ex5", 1, 2, 2, 2 ]
+];
+
+testQuery('SELECT * FROM test1 INNER JOIN test2 ON test1.x = 1',
```

Comment:
need test case for the optional use of the `INNER` keyword:

* `INNER JOIN` === `JOIN`

need test cases for the optional use of the `OUTER` keyword:

* `OUTER JOIN` === `FULL OUTER JOIN` === `FULL JOIN` 
* `RIGHT OUTER JOIN` === `RIGHT JOIN`
* `LEFT OUTER JOIN` === `LEFT JOIN`

need test cases for N-way joins (where N > 2 :)